node/src/types/deploy.rs

```diff
@@ -827,27 +828,380 @@ impl Deploy {
             let attempted = item
                 .args()
                 .get(ARG_AMOUNT)
-                .ok_or(DeployValidationFailure::MissingTransferAmount)?
+                .ok_or(DeployConfigurationFailure::MissingTransferAmount)?
                 .clone()
                 .into_t::<U512>()
-                .map_err(|_| DeployValidationFailure::InvalidTransferAmount)?;
+                .map_err(|_| DeployConfigurationFailure::InvalidTransferAmount)?;
             let minimum = U512::from(config.native_transfer_minimum_motes);
             if attempted < minimum {
-                return Err(DeployValidationFailure::InsufficientTransferAmount {
+                return Err(DeployConfigurationFailure::InsufficientTransferAmount {
                     minimum,
                     attempted,
                 });
             }
         }
 
-        self.is_valid()
+        Ok(())
     }
 
     #[cfg(test)]
     pub(crate) fn invalidate(&mut self) {
         self.header.chain_name.clear();
     }
 
+    #[cfg(test)]
+    pub(crate) fn valid_deploy(rng: &mut TestRng) -> Self {
```

Comment:
This method always creates an instance of native transfer. Maybe rename it to `valid_native_transfer` so that no one tries to use it when a non-transfer valid deploy is expected.

Reply:
Renamed as suggested